testsuite/tests/effects/test6.ml

```diff
@@ -8,15 +8,23 @@ type _ t += E : unit t
           | F : unit t
 
 let () =
-  let ok1 = ref false and ok2 = ref false in
-  let f r =
-    try perform E with Unhandled -> r := true in
-  f ok1;
+  let ok1 = ref false
+  and ok2 = ref true
+  and ok3 = ref false in
+  let f e r =
+    try perform e with
+    | Unhandled_effect E -> r := not !r
+  in
+  f E ok1;
   Printf.printf "%b\n%!" !ok1;
-  try_with f ok2 {
+
+  try f F ok2 with Unhandled_effect _ -> ();
```

Comment:
```suggestion
  begin try f F ok2 with Unhandled_effect _ -> () end;
```
Apologies. Forgot the parentheses!

Reply:
Somehow the original code still works, in an unintended and potentially confusing way :laughing:

Reply:
Indeed. Thanks.